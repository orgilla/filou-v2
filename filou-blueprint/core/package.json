{
  "name": "@filou/blueprint",
  "version": "1.5.4",
  "sideEffects": false,
  "main": "lib/cjs/index.js",
  "jsnext:main": "lib/es/index.js",
  "esnext": "lib/es/index.js",
  "module": "lib/es/index.js",
  "types": "lib/typings/index.d.ts",
  "babel": {
    "extends": "../../babelrc.js"
  },
  "author": "Orgilla <hi@orgil.la>",
  "contributors": [
    "Benjamin Kniffler <bkniffler@me.com>",
    "Philipp Achenbach <olymp@philipp-achenbach.de>"
  ],
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "scripts": {
    "build": "run-s compile",
    "compile": "run-p compile:*",
    "watch": "run-p \"compile:* --watch\"",
    "compile:es": "babel --out-dir ./lib/es --modules false --extensions .ts,.tsx ./src",
    "compile:cjs": "babel --out-dir ./lib/cjs --modules commonjs --extensions .ts,.tsx ./src",
    "compile:tsc": "tsc -b .",
    "bundle": "rollup -c ../../rollup.config.js"
  },
  "dependencies": {
    "@blueprintjs/core": "^3.8.0",
    "@blueprintjs/select": "^3.2.0",
    "@filou/core": "^1.5.4",
    "emotion": "^9.2.12"
  },
  "devDependencies": {
    "@babel/cli": "^7.1.5",
    "npm-run-all": "^4.1.3",
    "rimraf": "^2.6.2",
    "rollup": "^0.67.1",
    "typescript": "^3.1.6"
  }
}
