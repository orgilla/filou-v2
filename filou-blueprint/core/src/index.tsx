export * from '@blueprintjs/core';
export {
  Omnibar,
  MultiSelect,
  QueryList,
  Suggest,
  Select
} from '@blueprintjs/select';
export { default as Button } from './button';
export { default as ScreenLoader } from './screen-loader';
export { default as Tabs } from './tabs';
