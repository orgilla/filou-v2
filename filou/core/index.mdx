---
menu: Filou Components
name: Container
---

import { Playground, PropsTable } from 'docz';

## Container

## Examples

### Basic
