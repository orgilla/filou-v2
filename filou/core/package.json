{
  "name": "@filou/core",
  "version": "1.5.6",
  "sideEffects": false,
  "main": "lib/cjs/index.js",
  "jsnext:main": "lib/es/index.js",
  "esnext": "lib/es/index.js",
  "module": "lib/es/index.js",
  "types": "lib/typings/index.d.ts",
  "babel": {
    "extends": "../../babelrc.js"
  },
  "author": "Orgilla <hi@orgil.la>",
  "contributors": [
    "Benjamin Kniffler <bkniffler@me.com>",
    "Philipp Achenbach <olymp@philipp-achenbach.de>"
  ],
  "license": "MIT",
  "scripts": {
    "build": "run-s compile",
    "compile": "run-p compile:*",
    "watch": "run-p \"compile:* --watch\"",
    "compile:es": "babel --out-dir ./lib/es --modules false --extensions .ts,.tsx ./src",
    "compile:cjs": "babel --out-dir ./lib/cjs --modules commonjs --extensions .ts,.tsx ./src",
    "compile:tsc": "tsc -b .",
    "bundle": "rollup -c ../../rollup.config.js"
  },
  "dependencies": {
    "emotion": "^9.2.12",
    "fela": "^6.2.3",
    "fela-plugin-custom-property": "^7.0.5",
    "fela-plugin-embedded": "^5.2.13",
    "fela-plugin-extend": "^6.0.11",
    "fela-plugin-fallback-value": "^5.0.20",
    "fela-plugin-friendly-pseudo-class": "^5.0.18",
    "fela-plugin-named-keys": "^1.0.4",
    "fela-plugin-prefixer": "^5.0.21",
    "fela-plugin-unit": "^5.1.2",
    "polished": "^2.3.0",
    "tinycolor2": "^1.4.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.1.5",
    "npm-run-all": "^4.1.3",
    "rimraf": "^2.6.2",
    "rollup": "^0.67.1",
    "typescript": "^3.1.6"
  }
}
