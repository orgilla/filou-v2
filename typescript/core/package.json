{
  "version": "1.2.1",
  "name": "@filou/core",
  "description": "",
  "main": "lib",
  "esnext": "es",
  "module": "es",
  "types": "lib/index.d.js",
  "scripts": {
    "test": "jest",
    "clean": "rimraf lib es",
    "build": "yarn clean && tsc && tsc --module esnext --outDir es",
    "dev": "tsc --watch --module esnext --outDir es"
  },
  "author": "Orgilla <hi@orgil.la>",
  "contributors": [
    "Benjamin Kniffler <bkniffler@me.com>",
    "Philipp Achenbach <olymp@philipp-achenbach.de>"
  ],
  "dependencies": {
    "fela": "^6.2.1",
    "fela-plugin-custom-property": "^7.0.3",
    "fela-plugin-embedded": "^5.2.11",
    "fela-plugin-extend": "^6.0.9",
    "fela-plugin-fallback-value": "^5.0.18",
    "fela-plugin-friendly-pseudo-class": "^5.0.17",
    "fela-plugin-named-keys": "^1.0.3",
    "fela-plugin-prefixer": "^5.0.19",
    "fela-plugin-unit": "^5.0.17",
    "tinycolor2": "^1.4.1"
  },
  "peerDependencies": {
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "react-fela": "^8.0.0"
  },
  "license": "MIT",
  "gitHead": "a3049932b9c41c04e9b5b0b2b71eadeb8cc12e13"
}
