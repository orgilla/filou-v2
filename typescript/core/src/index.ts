export { default as FelaComponent } from './fela-component';
export { default as FelaStatic } from './fela-static';
export { default as ThemeProvider } from './theme-provider';
export { default as createRenderer } from './create-renderer';
export { default as StateProvider } from './state-provider';
export { default as Provider } from './provider';
export { default as Text } from './text';
export { default as If } from './if';
export * from './utils';
