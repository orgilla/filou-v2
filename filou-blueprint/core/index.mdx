---
menu: Filou Blueprint Components
name: Button
---

import { Playground, PropsTable } from 'docz';
import {
  DarkProvider,
  Provider,
  createRenderer,
  ThemeProvider
} from '@filou/core';
import '@blueprintjs/core/lib/css/blueprint.css';
import '@blueprintjs/select/lib/css/blueprint-select.css';
import { Button } from './src';
const renderer = createRenderer();

# Button

...

## Button

<PropsTable of={Button} />

## Examples

### Basic

<Playground>
  <Provider renderer={renderer}>
    <ThemeProvider theme={{ color: '#FBC02D', fontWeight: 300, dark: '#000' }}>
      <Button>Default</Button>
      <Button minimal intent="primary">
        Minimal Primary
      </Button>
      <Button minimal intent="danger">
        Minimal Danger
      </Button>
      <div style={{ backgroundColor: 'black' }}>
        <DarkProvider dark>
          <Button>Default</Button>
          <Button minimal intent="primary">
            Minimal Primary
          </Button>
          <Button minimal intent="danger">
            Minimal Danger
          </Button>
        </DarkProvider>
      </div>
    </ThemeProvider>
  </Provider>
</Playground>
