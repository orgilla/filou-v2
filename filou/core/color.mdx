---
menu: Getting Started
name: Theming & Colors
---

import { Playground, PropsTable } from 'docz';
import { Provider, FelaComponent, createRenderer } from '@filou/core';
import { Button } from '@filou/blueprint';
import materialColors from './src/material-colors';
import '@blueprintjs/core/lib/css/blueprint.css';
import '@blueprintjs/select/lib/css/blueprint-select.css';
const renderer = createRenderer();

# Theming

# Colors

## Usage

<Playground>
  <Provider renderer={renderer}>
    <Button>test</Button>
    <Button color={0}>test</Button>
    <Button color={0} minimal>
      test
    </Button>
  </Provider>
</Playground>

## Material Colors

<Playground>
  <Provider renderer={renderer}>
    <FelaComponent style={{ display: 'flex', flexWrap: 'wrap' }}>
      {materialColors.map((color, j) => (
        <FelaComponent
          style={{
            color: color.palette[7],
            textAlign: 'center',
            display: 'flex',
            flexDirection: 'column'
          }}
          color={color}
          key={color.palette[7]}
          render={({ className }) => (
            <div>
              <h4 className={className}>
                {j}: {color.color}
              </h4>
              {color.palette.map((palette, i) => (
                <FelaComponent
                  rule={({ theme, color, palette }) => ({
                    backgroundColor: color.palette[palette],
                    color: color.inverted[palette] ? theme.light : theme.dark,
                    padding: theme.space1,
                    margin: theme.space1,
                    minWidth: '10%'
                  })}
                  color={color}
                  palette={i}
                  key={color.palette[i]}
                  render={({ className }) => (
                    <div className={className}>
                      {i}: {palette}
                    </div>
                  )}
                />
              ))}
            </div>
          )}
        />
      ))}
    </FelaComponent>
  </Provider>
</Playground>
