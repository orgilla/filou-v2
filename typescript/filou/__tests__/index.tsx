import 'jest';
import * as React from 'react';
import * as test from 'react-test-renderer';
import { createRenderer, Provider } from '@filou/core';
import { renderToMarkup } from 'fela-dom';

describe('Grid', () => {
  it('should render correctly', () => {
    const renderer = createRenderer();
    const component = test.create(
      <Provider renderer={renderer}>
        <div />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(renderer)).toMatchSnapshot();
  });
});
