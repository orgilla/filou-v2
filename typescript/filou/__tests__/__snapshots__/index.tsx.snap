// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `<div />`;

exports[`Grid should render correctly 2`] = `""`;
